src/languages/typescript.ts

```diff
@@ -135,55 +147,68 @@ const nodeMatchers: Record<ScopeType, NodeMatcher> = {
     ["array"],
     (node) => isExpression(node) || node.type === "spread_element"
   ),
-  ifStatement: hasType("if_statement"),
-  class: possiblyExportedDeclaration("class_declaration", "class"),
-  statement: possiblyExportedDeclaration(...STATEMENT_TYPES),
-  arrowFunction: hasType("arrow_function"),
-  functionCall: hasType("call_expression", "new_expression"),
+  ifStatement: typeMatcher("if_statement"),
+  class: matcher(possiblyExportedDeclaration("class_declaration", "class")),
+  statement: matcher(possiblyExportedDeclaration(...STATEMENT_TYPES)),
+  arrowFunction: typeMatcher("arrow_function"),
+  functionCall: typeMatcher("call_expression", "new_expression"),
+  functionName: cascadingMatcher(
```

Comment:
Do you not also support `name` for typescript?

Reply:
It's using the one defined in `getPojoMatchers`. Unsure if it should stay there? Currently typescript is the only language using it, but it's the most generic implementation possible.

Reply:
Yeah I don't think it should be in `pojoMatchers`.  "Pojo" stands for "plain old javascript object", so more or less should only contain json stuff.  But if you want to do a rename / refactor I'm fine with it